functorch/_src/aot_autograd.py

```diff
@@ -409,14 +417,21 @@ def forward(ctx, *deduped_flat_tensor_args):
                 CompiledFunction.compiled_fw, deduped_flat_tensor_args
             )
             num_outs = CompiledFunction.num_outs
-            ctx.save_for_backward(*fw_outs[num_outs:])
+            num_symints = CompiledFunction.num_symints
+            if num_symints > 0:
```

Comment:
You don't actually need this condition, I think

Reply:
haha i do. but i want to rewrite this logic anyway.  i did not know this, but it turns out that `something[start:-0]` is not the same as `something[start:]`

Reply:
lol

Reply:
you could fix it with `something[start:len(something)-num_symints]`